tardis/model/base.py

```diff
@@ -562,8 +580,8 @@ def from_csvy(cls, config):
 
         dilution_factor = None
         if hasattr(csvy_model_data, 'columns'):
-            if 'w' in csvy_model_data.columns:
-                dilution_factor = csvy_model_data['w'].iloc[0:].to_numpy()
+            if 'dilution_factor' in csvy_model_data.columns:
```

Comment:
@wkerzendorf is this what you meant by "shifting the dilution factor into the code?"  This is the change that loads the given dilution factor profile into the csvy model.